File: app/src/main/java/me/foolishchow/androidplugins/fake/BaseGraph.java
package me.foolishchow.androidplugins.fake;

import androidx.annotation.IdRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.NavDestination;
import androidx.navigation.NavGraph;
import androidx.navigation.Navigator;
import androidx.navigation.fragment.NavHostFragment;

/**
 * Description:
 * Author: foolishchow
 * Date: 26/4/2021 8:49 PM
 */
public class BaseGraph extends NavGraph {

    public BaseGraph(NavController controller) {
        //noinspection unchecked
        super((Navigator<? extends NavGraph>) controller.getNavigatorProvider().getNavigator("navigation"));

        //setStartDestination();
        //new NavDestination(controller.getNavigatorProvider().getNavigator(""))
        //addDestination();
    }

    public void addAction(@IdRes int actionId,@IdRes int fromDestinationId,
                          @IdRes int targetDestinationId) {
    }

    public  <T extends Fragment> void addDestination(Class<T> tClass,int id,String label){

    }



}

File: app/src/main/java/me/foolishchow/autoparamdemo/fake/BaseGraph.kt
package me.foolishchow.autoparamdemo.fake

/**
 * Description:
 * Author: foolishchow
 * Date: 26/4/2021 8:49 PM
 */

//@Suppress("unchecked_cast")
//fun getNavigator(controller: NavController): Navigator<out NavGraph?> {
//    return controller.navigatorProvider.getNavigator<Navigator<*>>("navigation") as Navigator<out NavGraph?>
//}
//
//object NavigationUtils {
//    @JvmStatic
//    fun createNavigator(controller: NavController): NavGraph {
//        val navigator: Navigator<*> = controller.navigatorProvider.getNavigator("navigation")
//        return navigator.createDestination() as NavGraph
//    }
//
//    @JvmStatic
//    fun createDestination(controller: NavController): FragmentNavigator.Destination {
//        val navigator: FragmentNavigator = controller.navigatorProvider.getNavigator(FragmentNavigator::class.java)
//        return navigator.createDestination()
//    }
//
//}